content/articles/nesting-navigators-and-routing-with-react-navigation-in-react-native/index.md

```diff
@@ -0,0 +1,266 @@
+
+First, the `@react-navigation/native` and `@react-navigation/stack` and their peer dependencies should be installed.
+
+While revisiting our `App.js` we'll find that we added `NavigationContainer` and `createStackNavigator` to ease and allow navigation from one screen to another. We also added the two screens: `HomeScreen` and `NewWorkerScreen` that we will be navigating through and thus navigator is aware of your two screens.
+
+### Fourth Step - Passing Data to Other Screens Using Context
+
+It's possible to add new workers in `App.js`, but you'll want to do it in` NewWorkerScreen.js` so they appear in `HomeScreen.js` rather than `App.js`. Using React, you can easily integrate this feature into your interfaces. 
+
+As a beginning point let's add `NewWorkerContext` to the app.
+
+```
+nano NewWorkerContext.js
+```
+
+Exporting NewWorkerContex:
+
+```js
+import React from 'react';
+import { createAppContainer } from "react-navigation";
+import { createStackNavigator } from "react-navigation-stack";
+import { StyleSheet, Text, View } from 'react-native';
+
+export const NewWorkerContex = React.createContext();
+```
+
+The `NavigationContainer` is wrapped in a `Context.Provider` component so that any child in the component tree can listen for modifications to the context, and `NewWorkerContext` is created as a new Context object.
+
+After adding `NewWorkerContext` to the app we can also use it in our `HomeScreen` and` NewWorkerScreen`. If the currentWorkers or possibleWorkers list changes, the HomeScreen and NewWorkerContex will be able to refer to the new list.
+
+```js
+import 'react-native-gesture-handler';
+import React from "react";
+import { StyleSheet, Text, View, Button } from 'react-native';
+
+import HomeScreen from "./HomeScreen";
+import NewWorkerScreen from "./NewWorkerScreen";
+import { createAppContainer } from "react-navigation";
+import { createStackNavigator } from "react-navigation-stack";
+
+import { NewWorkerContex } from './NewWorkerContex';
+
+class HomeScreen extends React.Component {
+
+}
+
+HomeScreen.contextType = NewWorkerContex;
+```
+### Fifth Step - Adding Buttons to HomeScreen and FriendsScreen
+
+Add buttons to switch between the two screens, if necessary. As long as the navigation object is included in the StackNavigator, many of its useful attributes will be transmitted down to your screen.
+
+Add the following code in the HomeScreen.js to add a button and use the same way to another button in the NewWorkerScreen.js.
+
+```js
+<Button
+  title="Add new workers"
+  onPress={() =>
+    this.props.navigation.navigate('Wokers')
+  }
+/>
+
+```
+Now, in NewWorkerScreen.js, build a button for adding workers:
+
+```js
+{this.context.possibleWorkers.map((worker, index) => (
+  <Button
+    key={worker}
+    title={`Add ${worker}`}
+    onPress={() =>
+      this.context.addWorker(index)
+    }
+  />
+))}

```

Comment:
The context had no values passed down. What are PossibleWorkers? What does the addWorker function do? 
Please add these declarations and definitions to the context file.